app/assets/javascripts/discourse/app/components/post/cooked-html.gjs

```diff
@@ -47,48 +47,58 @@ export default class PostCookedHtml extends Component {
-          data: {
-            post: this.args.post,
-            cooked: this.cooked,
-            highlightTerm: this.highlightTerm,
-            isIgnored: this.isIgnored,
-            ignoredUsers: this.ignoredUsers,
-          },
-          createDetachedElement: this.#createDetachedElement,
-          currentUser: this.currentUser,
-          helper,
-          renderNestedPostCookedHtml: (
-            nestedElement,
-            nestedPost,
-            extraDecorators
-          ) => {
-            const nestedArguments = {
-              post: nestedPost,
-              streamElement: false,
+        try {
+          if (!this.#decoratorState.has(decorator)) {
+            this.#decoratorState.set(decorator, {});
+          }
+
+          const owner = getOwner(this);
+          const decorationCleanup = decorator(element, {
+            data: {
+              post: this.args.post,
+              cooked: this.cooked,
               highlightTerm: this.highlightTerm,
-              extraDecorators: [
-                ...this.extraDecorators,
-                ...makeArray(extraDecorators),
-              ],
-            };
-
-            helper.renderGlimmer(
+              isIgnored: this.isIgnored,
+              ignoredUsers: this.ignoredUsers,
+            },
+            createDetachedElement: this.#createDetachedElement,
+            currentUser: this.currentUser,
+            helper,
+            renderNestedPostCookedHtml: (
               nestedElement,
-              curryComponent(PostCookedHtml, nestedArguments, owner)
-            );
-          },
-          owner,
-          state: this.#decoratorState.get(decorator),
-        });
-
-        if (typeof decorationCleanup === "function") {
-          this.#pendingDecoratorCleanup.push(decorationCleanup);
+              nestedPost,
+              extraDecorators,
+              extraArguments
+            ) => {
+              const nestedArguments = {
+                ...extraArguments,
+                post: nestedPost,
+                streamElement: false,
+                highlightTerm: this.highlightTerm,
+                extraDecorators: [
+                  ...this.extraDecorators,
+                  ...makeArray(extraDecorators),
+                ],
+              };
+
+              helper.renderGlimmer(
+                nestedElement,
+                curryComponent(PostCookedHtml, nestedArguments, owner)
+              );
+            },
+            owner,
+            state: this.#decoratorState.get(decorator),
+          });
+
+          if (typeof decorationCleanup === "function") {
+            this.#pendingDecoratorCleanup.push(decorationCleanup);
+          }
+        } catch (e) {
+          // in case one of the decorators throws an error we want to surface it to the console but prevent
+          // the application from crashing
+
+          // eslint-disable-next-line no-console
+          console.error(e);
```

Comment:
Can we make it throw in qunit/system-spec environments?

Reply:
Done